controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
+        $this->ajaxDieMissingParams();
+    }
+
+    private function generateWishlistToken()
+    {
+        return strtoupper(substr(sha1(uniqid(rand(), true) . _COOKIE_KEY_ . $this->context->customer->id), 0, 16));
+    }
+
+    private function ajaxDieMissingParams()
+    {
+        $this->ajaxDie(
+            Tools::jsonEncode([
+                'success' => false,
+                'message' => $this->module->l('Request is missing one or multiple parameters', 'mywishlist')
+            ])
+        );
+    }
+
+    public function getProductsByWishlistAction($params)
+    {
+        $params['id_wishlist'] = 3;
+        $wishlistProducts = WishList::getProductByIdCustomer($params['id_wishlist'], $this->context->customer->id, $this->context->language->id);
+
+        if (empty($wishlistProducts)) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('No products found for this customer', 'mywishlist')
+                ])
+            );
+        }
+
+        $assembler = new ProductAssembler($this->context);
+        $presenterFactory = new ProductPresenterFactory($this->context);
+        $presentationSettings = $presenterFactory->getPresentationSettings();
+        $presenter = new ProductListingPresenter(
+            new ImageRetriever(
+                $this->context->link
+            ),
+            $this->context->link,
+            new PriceFormatter(),
+            new ProductColorsRetriever(),
+            $this->context->getTranslator()
+        );
+        $products_for_template = array();
+
+        if (is_array($wishlistProducts)) {
+            foreach ($wishlistProducts as $rawProduct) {
+                $products_for_template[] = $presenter->present(
+                    $presentationSettings,
+                    $assembler->assembleProduct($rawProduct),
+                    $this->context->language
+                );
+            }
+        }
+
+        $this->ajaxDie(
+            Tools::jsonEncode([
+                'status' => 'success',
+                'message' => $this->module->l('The list has been properly created', 'mywishlist'),
+                'datas' => [
+                    'products' => $products_for_template
+                ]
+            ])
+        );
+    }
+
+    public function addProductToCartAction($params)
+    {
+        $productAdd = WishList::addBoughtProduct(
+            $params['idWishlist'],
+            $params['id_product'],
+            $params['id_product_attribute'],
+            $params['id_cart'],
+            $params['quantity']
+        );
+        if($productAdd ===true) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'status' => 'success',
+                    'message' => $this->module->l('Product added to cart'),
+                ])
+            );
+        } else {
+            $this->ajaxDie(
+                Tools::jsonEncode([
```

Comment:
```suggestion
                json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007